### PR TITLE
tests(react): improve webchat tests

### PR DESCRIPTION
**Pull request by AinaVendrell**
## Description

- Fix `persistentMenu` test
- Add test for webchat messages 
- Create `expectToHaveRoles` and `expectNotToHaveRoles` to extract the common logic of the tests

## Context
`PersistentMenu` test was failing because the `closeLabel` button has no payload, path, webviews or url

## Approach taken / Explain the design
To solve the `PersistentMenu` error, check if the button has `onClick` prop

## Testing

The pull request...

- has unit tests
- has integration tests
- doesn't need tests because... **[provide a description]**

### CODE REVIEW

File: packages/botonic-react/tests/webchat/webchat.addmessages.test.jsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import {
+  renderUseWebchatHook,
+  expectToHaveRoles,
+  expectNotToHaveRoles,
+} from '../helpers/test-utils'
+import { render, screen } from '@testing-library/react'
+import { ROLES } from '../../src/constants'
+
+describe('TEST: webchat addMessageComponent', () => {
+  window.HTMLElement.prototype.scrollTo = function () {}
```
Comment (vanbasten17) [inline packages/botonic-react/tests/webchat/webchat.addmessages.test.jsx:14]: I would add a comment for why is this necessary

File: packages/botonic-react/tests/webchat/webchat.addmessages.test.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import { act } from 'react-test-renderer'
+import {
+  renderUseWebchatHook,
+  expectToHaveRoles,
+  expectNotToHaveRoles,
+} from '../helpers/test-utils'
+import { render, screen } from '@testing-library/react'
+import { ROLES } from '../../src/constants'
+
+describe('TEST: webchat addMessageComponent', () => {
```
Comment (vanbasten17) [inline packages/botonic-react/tests/webchat/webchat.addmessages.test.jsx:13]: describe(Adding webchat messageComponent) or similar for this group of tests

File: packages/botonic-react/tests/webchat/webchat.addmessages.test.jsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  window.HTMLElement.prototype.scrollTo = function () {}
+
+  it('When adding an Image message the webchat has StyledWebchat, MessageList, StyledMessage and ImageMssage', async () => {
```
Comment (vanbasten17) [inline packages/botonic-react/tests/webchat/webchat.addmessages.test.jsx:16]: it('TEST: When adding an Image message the webchat has StyledWebchat, MessageList, StyledMessage and ImageMessage'. <-- Mssage --> Message
Can you please put 'TEST:' for every "it" test?

File: packages/botonic-react/tests/helpers/test-utils.jsx [hunk 4]
```diff
@@ -53,3 +53,15 @@ export const createFile = ({ fakeData, fileName, mimeType, size }) => {
...
+  role.forEach(r => {
+    expect(screen.queryByRole(r)).toBeTruthy()
+  })
+}
+
+export function expectNotToHaveRoles(role, screen) {
+  role.forEach(r => {
+    expect(screen.queryByRole(r)).toBeNull()
+  })
+}
```
Comment (vanbasten17) [inline packages/botonic-react/tests/helpers/test-utils.jsx:67]: amazing!! 💯 💯 💯

**Review by vanbasten17 — APPROVED** [review]
Looking good Aina, good work! 👏 Can you please rename tests literals to contain 'TEST:' ?

File: packages/botonic-react/src/constants.js [hunk 5]
```diff
@@ -155,4 +155,11 @@ export const ROLES = {
   TRIGGER_BUTTON: 'trigger-button',
   TYPING_INDICATOR: 'typing-indicator',
   TEXT_BOX: 'textbox',
+  WEBVIEW: 'webview',
+  WEBVIEW_HEADER: 'webview_header',
+  STYLED_MESSAGE: 'styled-message',
```
Comment (elozano98) [inline packages/botonic-react/src/constants.js:160]: If `STYLED_MESSAGE`, `IMAGE_MESSAGE`, `AUDIO_MESSAGE`, `VIDEO_MESSAGE` and `DOCUMENT_MESSAGE` are used to specify the type of message (not the role) maybe they should be defined in a new constant named `MESSAGE_TYPE`.

File: packages/botonic-react/tests/helpers/test-utils.jsx (hunk 4 above)
Comment (elozano98) [reply packages/botonic-react/tests/helpers/test-utils.jsx:67]: good job!! :+1:

**Review by elozano98 — APPROVED** [review]
Good job with these test improvements! :clap: :100:

File: packages/botonic-react/src/constants.js (hunk 5 above)
Comment (vanbasten17) [reply packages/botonic-react/src/constants.js:160]: in fact, i suggest not having references of the style: 'styled-whatever', so we should have the role 'message', 'webchat', ...

File: packages/botonic-react/src/constants.js [hunk 6]
```diff
@@ -155,4 +155,11 @@ export const ROLES = {
   TRIGGER_BUTTON: 'trigger-button',
   TYPING_INDICATOR: 'typing-indicator',
   TEXT_BOX: 'textbox',
+  WEBVIEW: 'webview',
+  WEBVIEW_HEADER: 'webview_header',
```
Comment (vanbasten17) [inline packages/botonic-react/src/constants.js:159]: be consistent: 'webview-header'

**Review by vanbasten17 — CHANGES_REQUESTED** [review]

**Review by dpinol — APPROVED** [review]
